src/index/Index.cpp

```diff
@@ -1287,11 +1287,11 @@ size_t Index::getHasPredicateFullSize() const {
 }
 
 // _____________________________________________________________________________
-string Index::idToString(Id id) const {
+std::optional<string> Index::idToOptionalString(Id id) const {
   if (id < _vocab.size()) {
     return _vocab[id];
   } else if (id == ID_NO_VALUE) {
-    return "";
+    return std::nullopt;
   } else {
     id -= _vocab.size();
     AD_CHECK(id < _vocab.getExternalVocab().size()) {
```

Comment:
Why is this a method of the Index and not of the Vocabulary?

Reply:
Good question

Reply:
Yes but we can still implement it it the Vocabulary and wrap it in the index
` Index::getOptionalForId(id) {return _vocab.getOptionalForId;}`

In general, the Index class is already full enough (functionality AND Code duplications:))